ui/src/infra-model/view/dialogs/new-author-dialog.tsx

```diff
@@ -14,38 +14,54 @@ import * as Snackbar from 'geoviite-design-lib/snackbar/snackbar';
 import { isEqualWithoutWhitespace } from 'utils/string-utils';
 
 type NewAuthorDialogProps = {
-    authors?: Author[];
+    nameSuggestion: string | undefined;
+    authors: Author[];
     onClose: () => void;
     onSave: (author: Author) => void;
 };
 
-export const NewAuthorDialog: React.FC<NewAuthorDialogProps> = ({ authors, onClose, onSave }) => {
+export const NewAuthorDialog: React.FC<NewAuthorDialogProps> = ({
+    nameSuggestion,
+    authors,
+    onClose,
+    onSave,
+}) => {
     const { t } = useTranslation();
-    const [authorName, setAuthorName] = React.useState<string>('');
+    const [authorName, setAuthorName] = React.useState<string>(nameSuggestion || '');
     const [canSave, setCanSave] = React.useState<boolean>(false);
     const [duplicateName, setDuplicateName] = React.useState<boolean>(false);
     const [saveInProgress, setSaveInProgress] = React.useState<boolean>(false);
+    const nameFieldRef = React.useRef<HTMLInputElement>(null);
 
-    const debouncer = React.useCallback(
-        debounce((newName: string) => {
-            const existingAuthor = authors?.find((author) =>
-                isEqualWithoutWhitespace(author.companyName, newName),
-            );
+    function validateName(newName: string): void {
+        const existingAuthor = authors?.find((author) =>
+            isEqualWithoutWhitespace(author.companyName, newName),
+        );
+
+        if (existingAuthor) setDuplicateName(true);
+        else if (newName) setCanSave(true);
+    }
```

Comment:
Erotettu omaksi funktiokseen debouncen sisältä, että mahdollinen annettu nimiehdotus voidaan validoida välittömästi

Reply:
Joo samaa mieltä tuo on aika meh näköistä logiikkaa. Lisäksi näyttää erikoiselta että tuolta alempaa (onNameChange) settaillaan näitä samoja bittejä ja sitten täällä settaillaan lisää. Tuo menee todella hähmäiseksi seurata ja debounce tekee tuosta kahdessa paikassa settailusta vielä oudompaa koska niiden väliin tulee pieni ajanhetkikin. 

Jos tuota ei haluakaan rempata tilattomammaksi niin sen voisi ainakin tehdä että ne asetetaan yhdessä paikassa, esim niin että täällä laitetaan arvo aina ( `setDuplicateName(existingAuthor)` jne) ja kutsujat jättää ne rauhaan. Toki vähän jännää sekin että validateName-niminen funkkari oikeasti settailee jotain, mutta se toki tulee juuri tuosta samasta että se validoinnin tulos pyöritellään tilamuuttujien kautta.